build.rs

```diff
@@ -1,53 +1,161 @@
+    url.hash(&mut hasher);
+    let hash = hasher.finish();
+    format!("{hash}_machdxcompiler")
+}
+
+/// Generates C API bindings.
+fn generate_bindings() {
+    let bindings = bindgen::Builder::default()
+        // The input header we would like to generate
+        // bindings for.
+        .header("mach_dxc.h")
+        // Finish the builder and generate the bindings.
+        .generate()
+        // Unwrap the Result and panic on failure.
+        .expect("Unable to generate bindings");
+
+    // Write the bindings to the src/bindings.rs file.
+    let out_path =
+        PathBuf::from(env::var("OUT_DIR").expect("Failed to get OUT_DIR environment variable"));
+    bindings
+        .write_to_file(out_path.join("cbindings.rs"))
+        .expect("Couldn't write bindings!");
+}
+
 /// Gets the URL from which the DXC binary should be downloaded.
-fn get_target_url(target: &str, static_crt: bool) -> String {
-    const BASE_URL: &str = "https://github.com/DouglasDwyer/mach-dxcompiler/releases/download/2024.11.22%2Bbfceb9a.1/";
+fn get_target_url(static_crt: bool) -> String {
+    const BASE_URL: &str = "https://github.com/DouglasDwyer/mach-dxcompiler/releases/download";
+    const LASTEST_MSVC_RELEASE: &str = "2024.11.22+bfceb9a.1";
+    const LASTEST_OTHER_RELEASE: &str = "2024.11.22+df583a3.1";
+    const AVAILABLE_TARGETS: &[&str] = &[
+        "x86_64-linux-gnu",
+        "x86_64-linux-musl",
+        "aarch64-linux-gnu",
+        "aarch64-linux-musl",
+        "x86_64-windows-gnu",
+        "x86_64-windows-msvc",
+        "aarch64-windows-gnu",
+        "x86_64-macos-none",
+        "aarch64-macos-none",
+    ];
+    let arch = env::var("CARGO_CFG_TARGET_ARCH").expect("Failed to get architecture");
+    let os = env::var("CARGO_CFG_TARGET_OS").expect("Failed to get os");
+    // CARGO_CFG_TARGET_ENV may be empty
+    let mut abi = env::var("CARGO_CFG_TARGET_ENV").unwrap_or_default();
+    if abi.is_empty() {
+        abi = "none".to_owned();
+    }
+    let target = format!("{arch}-{os}-{abi}");
+
+    if !AVAILABLE_TARGETS.contains(&target.as_str()) {
+        panic!("Unsupported target: {target}\nCheck supported targets on {BASE_URL}");
+    }
+    let is_msvc = abi == "msvc";
+    let release = if is_msvc {
+        LASTEST_MSVC_RELEASE
+    } else {
+        LASTEST_OTHER_RELEASE
+    };
+    let crt = if is_msvc && static_crt {
+        "Dynamic_lib"
+    } else {
+        "lib"
+    };
+    format!("{BASE_URL}/{release}/{target}_ReleaseFast_{crt}.tar.gz")
+}
 
-    BASE_URL.to_string() + match (target, static_crt) {
-        ("x86_64-pc-windows-msvc", false) => "x86_64-windows-msvc_ReleaseFast_Dynamic_lib.tar.gz",
-        ("x86_64-pc-windows-msvc", true) => "x86_64-windows-msvc_ReleaseFast_lib.tar.gz",
-        ("x86_64-pc-windows-gnu", _) => "x86_64-windows-gnu_ReleaseFast_lib.tar.gz",
-        _ => panic!("Unsupported target '{target}' for mach-dxcompiler")
+fn is_file_exists(file_path: &Path) -> bool {
```

Comment:
Would it be possible to add a purpose statement in a doc-comment above this function?

Reply:
Forgot to add :/